Updated migration guidelines for 2020.02.00 (#5286)
geosolutions-it/MapStore2: docs/developer-guide/mapstore-migration-guide.md

# General Migration Guidelines

Updating MapStore could mean:

- updating an existing installation
- updating a MapStore project created for a previous version

To update an existing installation you usually have to do nothing except eventually to execute queries on your database to update the changes to the database schema. 

> In case of a project the it becomes a little more complicated, depending on the customization.

Reviewer: ```suggestion
In case of a project it becomes a little more complicated, depending on the customization.
```

This is a list of things to check if you want to update from a previous version valid for every version.

- Take a list to migration notes below for your version
- Take a look to the release notes
- update your `package.json` to latest libs versions
- take a look at your custom files to see if there are some changes (e.g. `localConfig.js`, `proxy.properties`)
- Optionally check also accessory files like `.eslinrc`, if you want to keep aligned with lint standards.
- Follow the instructions below, in order, from your version to the one you want to update to.

## Migration from 2020.01.00 to 2020.02.00

### Translation files

- The translations file extension has been changed into JSON. Now translation files has been renamed from `data.<locale>` to `data.<locale>.json`. This change makes the `.json` extension mandatory for all translation files. This means that depending projects with custom translation files should be renabled in the same name. E.g. `data.it-IT` have to be renamed as `data.it-IT.json`

### Database Update

Database schema has changed. To update your database you need to apply this SQL scripts to your database

- Update the user schema

> tun the script available [here](https://github.com/geosolutions-it/geostore/tree/master/doc/sql/migration/postgresql):

Reviewer: ```suggestion
run the script available [here](https://github.com/geosolutions-it/geostore/tree/master/doc/sql/migration/postgresql):
```

For more info see the related [issue](https://github.com/geosolutions-it/MapStore2/issues/4569)

Moreover a new category has been added for future features, called GEOSTORY.

> you can add it by running on the database the following script

Author: ```suggestion
It is not necessary for this release, but, to follow the update sequence, you can add it by executing the following line.
```
